core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -601,6 +663,14 @@ public boolean isDefined() {
       return isDefined;
     }
 
+    /**
+     * Returns index for key.
```

Comment:
I think in the `HashedRelation` we don't care too much about how the key index is generated, and just require the index to be unique.

How about `Returns a unique index of the key.`